test/libsolidity/util/compiler/InternalCompiler.h

```diff
@@ -0,0 +1,70 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <liblangutil/SourceReferenceFormatter.h>
+
+#include <test/libsolidity/util/compiler/Compiler.h>
+
+#include <test/libsolidity/util/SoltestErrors.h>
+
+using namespace solidity::langutil;
+
+namespace solidity::frontend::test
+{
+
+using MappedContracts = std::map<std::string, std::vector<CompiledContract>>;
+using Errors = ErrorList;
+
+/**
+ * Abstraction of the internal compiler aka. `CompilerStack`.
```

Comment:
```suggestion
 * Abstraction of the `CompilerStack`.
```

abstraction of the internal internal compiler stack :D